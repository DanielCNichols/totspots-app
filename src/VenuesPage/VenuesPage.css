.VenuesPage {
  width: 90%;
  margin: 0 auto 100px auto;
}

.VenuesPage header {
  margin: 30px 0;
}

@media screen and (min-width: 450px) {
  .VenuesPage {
    width: 75%;
    margin: 100px auto;
  }
}

@media screen and (min-width: 800px) {
  .VenuesPage {
    min-width: 250px;
    width : calc(50% - 220px); 
    width: 50%;
    float: right;
    list-style: none;
    margin-right: 200px;
    margin-top: 100px;
    padding-left: 100px;
  }
}
