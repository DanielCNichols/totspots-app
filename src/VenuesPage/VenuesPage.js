import React, { useEffect, useState } from 'react';
import { TsReview, GoogleReview } from '../Review/Review';
import s from './VenuesPage.module.css';
import InfoCard from '../InfoCard/InfoCard';
import ApiService from '../services/api-service';
import { detail } from '../reference';
import config from '../config';
import Rating from '../Rating/Rating';
import ReviewForm from '../ReviewForm/ReviewForm';
import { FaStar, FaChild } from 'react-icons/fa';
import FormContainer from '../FormContainer/FormContainer';
import { MdRateReview, MdFavorite } from 'react-icons/md';

import PhotoElement from '../PhotoElement/PhotoElement';

//! If mobile, render the review form as an added div. Else, render it as a Modal.
//Todo: Refactor and export modal from the other thing.

const VenuesPage = props => {
  const [venue, setVenue] = useState({});
  const [tsReviews, setTSReviews] = useState(true);
  const [showReviewForm, setShowReviewForm] = useState(false);
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    // ApiService.getVenueDetails(props.match.params.id)
    //   .then(venue => {
    setVenue(detail);
    setLoading(false);
    // })
    //   .catch(err => setError(err));

    // setVenue(detail);
    // setLoading(false);
  }, []);

  function handleToggleReviewForm() {
    setShowReviewForm(!showReviewForm);
  }

  function handleToggleReviews() {
    setTSReviews(!tsReviews);
  }

  function renderTypes() {
    let { types } = venue.result;
    let formatted = types.map(t =>
      t !== 'establishment' && t !== 'point_of_interest'
        ? t.split('_').join(' ')
        : ''
    );

    return formatted.map((t, idx) => {
      return t.length ? (
        <li
          style={{
            display: 'inline-block',
            textTransform: 'capitalize',
          }}
          key={idx}
        >
          {t}
        </li>
      ) : null;
    });
  }

  return (
    <section className={s.venuePageContainer}>
      {loading ? (
        <p>Loading...</p>
      ) : (
        <>
          <div className={s.hero}>
            <img
              src={`https://maps.googleapis.com/maps/api/place/photo?maxwidth=1000&photoreference=${venue.result.photos[0].photo_reference}&key=${config.GKEY}`}
              alt="venue front"
            />
          </div>
          <div className={s.venueContainer}>
            <div className={s.venueInfo}>
              <h2>{venue.result.name}</h2>
              <Rating
                value={venue.result.rating}
                symbol={FaStar}
                iconClass="star"
              />
              <div className={s.types}>{renderTypes()}</div>
              <ul className={s.features}>
                {venue.amenities.map(amenity => {
                  return <li key={amenity.id}>{amenity.amenity_name}</li>;
                })}
              </ul>
              <div className={s.ratings}>
                <div className={s.ratingElement}>
                  <p>Avg. Totspots Rating:</p>
                  <Rating
                    value={venue.tsAverages.avgrating}
                    symbol={FaChild}
                    iconClass="totspots"
                  />
                </div>
              </div>
              <div className={s.businessControls}>
                <button>
                  <MdFavorite className={s.addFavorite} />
                  <p>Add to favorites</p>
                </button>
              </div>
            </div>

            <InfoCard venue={venue.result} />

            <div className={s.photos}>
              <div>
                <h3>Photos</h3>
                <PhotoElement photos={venue.result.photos} />
              </div>
            </div>

            <FormContainer />
            <div className={s.reviewControls}>
              <button onClick={() => handleToggleReviewForm()}>
                <MdRateReview className={s.addReview} />
                <p>Add a Review</p>
              </button>
              <button onClick={() => handleToggleReviews()}>Toggel</button>
            </div>

            <div className={s.reviews}>
              <div className={s.reviewFormContainer}>
                {showReviewForm && (
                  <ReviewForm
                    venue_id={props.match.params.id}
                    closeForm={handleToggleReviewForm}
                  />
                )}
              </div>

              {tsReviews && !showReviewForm && (
                <>
                  <h3>Here's what people are saying: </h3>
                  {venue.tsReviews ? (
                    venue.tsReviews.map(review => {
                      return <TsReview key={review.id} review={review} />;
                    })
                  ) : (
                    <p>Looks like there are no reviews... yet!</p>
                  )}
                </>
              )}

              {!tsReviews &&
                !showReviewForm(
                  <>
                    {venue.result.reviews.map((review, idx) => {
                      return <GoogleReview key={idx} review={review} />;
                    })}
                  </>
                )}
            </div>
          </div>
        </>
      )}
    </section>
  );
};

export default VenuesPage;
