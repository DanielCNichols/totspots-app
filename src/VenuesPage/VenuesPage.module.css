* {
  box-sizing: border-box;
}
/* Todo: Mobile Styles */
.hero {
  width: 100%;
  background-color: gray;
  display: flex;
  height: 200px;
  margin-bottom: 25px;
}

.hero img {
  width: 100%;
  height: auto;
  object-fit: cover;
  object-position: center;
}

.venuePageContainer {
  width: 100%;
  margin: 60px auto;
}

.venueContainer {
  width: 100%;
  padding: 0 5px;
}

.venueInfo {
  width: 100%;
  margin: 0 auto;
  text-align: left;
}

.venueInfo h2 {
  margin: 10px 0;
  text-align: left;
  font-size: 1.5em;
}

.types {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}
.types li {
  margin-right: 3px;
}

.types li::before {
  content: '|';
  margin: 0 3px;
}

.types li:first-of-type::before {
  content: none;
}

.features {
  display: flex;
  flex-wrap: wrap;
  text-align: left;
  justify-content: space-between;
  flex: 1 1 160px;
}

.features li {
  margin: 0 5px 5px 0;
}

.features li::before {
  content: '+';
}

.ratings {
  margin: 15px 0;
}
.ratingElement {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 10px 0;
  padding: 0 10px 0 0;
}

.businessControls {
  display: flex;
  justify-content: space-around;
  margin-bottom: 10px;
}

.businessControls button {
  width: 45%;
  height: auto;
  padding: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: transparent;
  border: none;
}

.addFavorite {
  color: #f31212;
}

.addReview {
  color: #1f2464;
}

.businessControls svg {
  font-size: 2em;
}

.businessControls button p {
  width: 50%;
  word-wrap: break-word;
}

.photos {
  width: 100%;
}

.photos h3 {
  margin-bottom: 10px;
}

.photoContainer {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 5px;
}

.photoContainer img {
  width: 100%;
  height: auto;
  object-fit: cover;
  object-position: center;
}

.tabNav {
  position: relative;
  display: flex;
  right: 10px;
  justify-content: space-between;
  align-items: center;
  margin: 20px auto;
  width: 105%;
  background-color: white;
  border-top: 2px solid #e7e7e7;
  border-bottom: 2px solid #e7e7e7;
  height: 40px;
}

/* Todo: Tablet Display */

/* Todo: Desktop */

@media screen and (min-width: 800px) {
  .venuePageContainer {
    margin: 50px auto;
  }
  .hero {
    height: 300px;
    /* /* position: absolute; */
    width: 100%;
    height: 350px;
    background-color: gray;
    display: flex;
  }

  .hero img {
    width: 100%;
    object-fit: cover;
    object-position: center;
  }

  .venueContainer {
    width: 70%;
    margin: 25px auto;
    display: grid;
    grid-template-columns: 3fr 2fr;
    grid-template-rows: auto 1fr;
    grid-column-gap: 20%;
    padding: 20px;
  }

  .venueInfo {
    width: 100%;
    margin: 0;
    padding: 20px 20px 20px 0;
  }

  .venueHeader h2 {
    font-size: 2em;
  }

  .features {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 20px 20px;
  }

  .reviews {
    grid-column-start: 1;
    grid-row-start: 3;
  }
}
