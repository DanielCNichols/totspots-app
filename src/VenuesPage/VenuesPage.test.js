
import React from 'react';
import ReactDOM from 'react-dom'
import VenuesPage from './VenuesPage.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders VenuesPage without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <VenuesPage />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders VenuesPage as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <VenuesPage name="VenuesPage"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
