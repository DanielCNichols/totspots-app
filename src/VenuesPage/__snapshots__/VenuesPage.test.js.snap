// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VenuesPage as expected 1`] = `
<section
  className="VenuesPage"
>
  <section
    className="venueProfile"
  >
    <header>
      <h2 />
    </header>
    <div
      className="contact"
    >
      <p
        className="type"
      >
        Type: 
      </p>
      <p />
      <span
        className="city"
      />
      , 
      <span
        className="state"
      />
      <span />
      <p />
      <p />
    </div>
    <div
      className="ratings"
    >
      <div
        className="average"
      >
        Overall Rating: 
        <span
          className="rating"
        >
          <span
            className="NA"
          >
            No reviews
          </span>
        </span>
      </div>
      <div
        className="average"
      >
        Overall Price: 
        <span
          className="price"
        >
          <span
            className="NA"
          >
            Price unavailable
          </span>
        </span>
      </div>
      <div
        className="average"
      >
        Overall Volume Level: 
        <span
          className="volume"
        >
          <span
            className="NA"
          >
            volume unavailable
          </span>
        </span>
      </div>
      <div
        className="amenities"
      >
        <h3>
          Features
        </h3>
      </div>
    </div>
    <div
      className="venue_controls"
    >
      <button
        onClick={[Function]}
      >
        Add a review
      </button>
      <button
        onClick={[Function]}
      >
        Add to Saved Venues
      </button>
    </div>
  </section>
  <header>
    <h3>
      Here's what people are saying: 
    </h3>
  </header>
  <div>
    <p>
      Be the first to review this venue!
    </p>
  </div>
</section>
`;
